sql/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -420,8 +422,8 @@ public Symbol visitFunction(Function function, Context context) {
             } else if (Operators.LOGICAL_OPERATORS.contains(functionName)) {
                 boolean proxyBelowPre = context.proxyBelow;
                 boolean proxyBelowPost = proxyBelowPre;
-                ArrayList<Symbol> args = new ArrayList<>(function.arguments().size());
-                for (Symbol arg : function.arguments()) {
+                ArrayList<Symbol> args = new ArrayList<>(arguments.size());
```

Comment:
maybe name this `newArgs`, given that there is now a `arguments` local variable the difference between `args` and `arguments` is otherwise less clear.